dev-guide/cluster-fleet-evaluation.md

```diff
@@ -0,0 +1,181 @@
+---
+title: cluster-fleet-evaluation
+authors:
+  - "@aravindhp"
+reviewers:
+  - "@openshift/openshift-patch-managers"
+approvers:
+  - "@openshift/openshift-patch-managers"
+creation-date: 2022-08-16
+last-updated: 2022-08-16
+---
+
+# Cluster Fleet Evaluation
+
+## Problem Statement
+
+OpenShift components sometimes need to introduce a change that could result in
+highly visible alerts, breakages or upgrade failures on user clusters. In
+scenarios like this it will be useful for the component owner to assess how many
+customers would be affected before rolling out the change and based on that
+feedback decide on the level of investment required to properly inform the
+customer base, or change implementation details.
+
+## Solution
+
+Rather than introducing the change, add just the detection logic to the operator
+for the change and associate it with a
+[ClusterStatusConditionTypes](https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L145)
+called [EvaluationConditionDetected](https://github.com/openshift/api/pull/1250).
+If the
+[ConditionStatus](https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L124)
+associated with this type is
+[ConditionTrue](https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L107)
+and the includes the
+[Reason]([Reason](https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L133))
+you are expecting, it will indicate that the change would have affected that
+cluster. However, if the _ConditionStatus_ is _false_, or it's _true_ but the
+_Reason_ does not include the _Reason_ you are expecting, the change doesn't
+affect the cluster.  You can join multiple _Reason_ to test multiple scenarios.
+
+Now release the detection logic to master and
+[backport](https://docs.google.com/document/d/1PC87sSFa_zGCk95kXDW-wrVxnlgBmkHqpOgQnd4bbUw/edit)
+it to a N-1 Z stream release where N is the current release under development.
+The Z stream backport is useful as it allows the team to gather the data without
+going through a full release cycle and waiting for customers to upgrade to newer
+minor versions. However, we still have to wait for them to upgrade to a patch
+level that includes the new info gathering logic.
+
+How far back the backports should be done is subjective and the teams
+should aim to backport as far back as possible so as to collect a reasonable
+amount data that allows them to reach a conclusion. This decision will be
+largely based on what versions the bulk of the fleet of clusters are running.
```

Comment:
I thought we said we wanted at least back to the oldest supported EUS release, did we change that?

Reply:
It could still mean the oldest supported EUS release but for some changes it might not be possible. So I wrote it so that the team involved can make that decision.

Reply:
I'm fine with this being subjective but maybe add a bit that explains the problem. Clusters running 4.8.z within 90-120 days of GA are skewed toward younger clusters which may not illustrate the same baggage that clusters upgrading to 4.8 today in order to avoid 4.6 and 4.7 EOL dates may have. Consider that in context of problem domain.

Reply:
```suggestion
largely based on what versions the bulk of the fleet of clusters are running
and the nature of the problem such as whether or not the problem domain
is likely influenced by cluster age or the version it was installed with. For
instance clusters running 4.8 today, 12 months after GA, are disproportionately
older and installed on older versions than those which were running 4.8 in
the first 90 days after GA.
```

Reply:
That makes sense @sdodson. I have added your suggestion.